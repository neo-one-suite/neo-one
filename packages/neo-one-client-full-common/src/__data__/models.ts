import { common, ContractParameterTypeModel, ECPoint, SignatureString, UInt160 } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';
import {
  ContractABIModel,
  ContractEventDescriptorModel,
  ContractFeaturesModel,
  ContractGroupModel,
  ContractManifestModel,
  ContractMethodDescriptorModel,
  ContractParameterDefinitionModel,
  ContractPermissionDescriptorModel,
  ContractPermissionModel,
  ContractStateModel,
} from '../models';

export const contractParamDefinitionModel = {
  any: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Any,
    name: 'param',
  }),
  boolean: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Boolean,
    name: 'param',
  }),
  byteArray: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.ByteArray,
    name: 'param',
  }),
  hash160: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Hash160,
    name: 'param',
  }),
  hash256: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Hash256,
    name: 'param',
  }),
  array: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Array,
    name: 'param',
  }),
  integer: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Integer,
    name: 'param',
  }),
  interopInterface: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.InteropInterface,
    name: 'param',
  }),
  map: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Map,
    name: 'param',
  }),
  publicKey: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.PublicKey,
    name: 'param',
  }),
  signature: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Signature,
    name: 'param',
  }),
  string: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.String,
    name: 'param',
  }),
  void: new ContractParameterDefinitionModel({
    type: ContractParameterTypeModel.Void,
    name: 'param',
  }),
};

export const contractMethodDescriptorModel = (
  parameters: readonly ContractParameterDefinitionModel[] = [contractParamDefinitionModel.boolean],
  returnType: ContractParameterTypeModel = ContractParameterTypeModel.Void,
  name = 'function',
  offset = 0,
) => new ContractMethodDescriptorModel({ name, parameters, returnType, offset });

export const contractEventDescriptorModel = (
  parameters: readonly ContractParameterDefinitionModel[] = [contractParamDefinitionModel.boolean],
  name = 'event',
) => new ContractEventDescriptorModel({ name, parameters });

export const contractAbiModel = (
  methods: readonly ContractMethodDescriptorModel[] = [contractMethodDescriptorModel()],
  events: readonly ContractEventDescriptorModel[] = [contractEventDescriptorModel()],
  hash = common.bufferToUInt160(Buffer.alloc(20, 1)),
) => new ContractABIModel({ hash, methods, events });

export const contractGroupModel = (
  publicKey: ECPoint = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  signature: Buffer = Buffer.from(
    'ccaab040cc25021c91567b75db4778853441869157b8f6aad960cdcf1069812480027a528ca9b98e2205027de20696f848cf81824eeb7af1d5110870870ceb67',
    'hex',
  ),
) => new ContractGroupModel({ publicKey, signature });

export const contractPermissionDescriptorModel = (hashOrGroupType: 'uint160' | 'ecpoint' | undefined) => {
  let hashOrGroup: UInt160 | ECPoint | undefined;
  if (hashOrGroupType === 'uint160') {
    hashOrGroup = common.bufferToUInt160(Buffer.alloc(20, 1));
  } else if (hashOrGroupType === 'ecpoint') {
    hashOrGroup = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  }

  return new ContractPermissionDescriptorModel({ hashOrGroup });
};

export const contractPermissionModel = (
  hashOrGroupType: 'uint160' | 'ecpoint' | undefined,
  methods: readonly string[] = [],
) => new ContractPermissionModel({ contract: contractPermissionDescriptorModel(hashOrGroupType), methods });

export const contractManifestModel = (
  groups: readonly ContractGroupModel[] = [contractGroupModel()],
  features: ContractFeaturesModel = ContractFeaturesModel.HasStoragePayable,
  abi: ContractABIModel = contractAbiModel(),
  permissions: readonly ContractPermissionModel[] = [contractPermissionModel('uint160', ['method1'])],
  trusts: readonly UInt160[] = [common.bufferToUInt160(Buffer.alloc(20, 1))],
  safeMethods: readonly string[] = ['method1', 'method2'],
  supportedStandards: readonly string[] = [],
) => new ContractManifestModel({ groups, features, supportedStandards, abi, permissions, trusts, safeMethods });

export const contractModel = (
  id = 1,
  script: Buffer = Buffer.alloc(25),
  manifest: ContractManifestModel = contractManifestModel(),
) => new ContractStateModel({ id, script, manifest });
