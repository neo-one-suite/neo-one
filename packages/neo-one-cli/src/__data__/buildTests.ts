/// <reference types="@neo-one/build-tools/types/e2e" />
// tslint:disable no-any
import { common, Contract, ContractABI, crypto, privateKeyToAddress, UserAccountID } from '@neo-one/client-common';
import { Client, DeveloperClient, Hash256 } from '@neo-one/client-core';
import { constants } from '@neo-one/utils';
import BigNumber from 'bignumber.js';
import * as nodePath from 'path';
import { getClients } from './getClients';
import { getContracts } from './getContracts';

const TO_PRIVATE_KEY = '7d128a6d096f0c14c3a25a2b0c41cf79661bfcb4a8cc95aaaea28bde4d732344';
const TO_PUBLIC_KEY = '02028a99826edc0c97d18e22b6932373d908d323aa7f92656a77ec26e8861699ef';
crypto.addPublicKey(common.stringToPrivateKey(TO_PRIVATE_KEY), common.stringToECPoint(TO_PUBLIC_KEY));

const verifyICOContract = (contract?: Contract): void => {
  expect(contract).toBeDefined();
  if (contract === undefined) {
    throw new Error('For TS');
  }
};

const verifyTokenContract = (contract?: Contract): void => {
  expect(contract).toBeDefined();
  if (contract === undefined) {
    throw new Error('For TS');
  }
};

const verifyEscrowContract = (contract?: Contract): void => {
  expect(contract).toBeDefined();
  if (contract === undefined) {
    throw new Error('For TS');
  }
};

const mintTokens = async (ico: any, accountID: UserAccountID, developerClient?: DeveloperClient): Promise<void> => {
  const mintResult = await ico.mintTokens({
    sendTo: [
      {
        amount: new BigNumber(10),
        asset: Hash256.NEO,
      },
    ],
  });

  const [mintReceipt] = await Promise.all([
    mintResult.confirmed(),
    developerClient === undefined ? Promise.resolve() : developerClient.runConsensusNow(),
  ]);
  if (mintReceipt.result.state === 'FAULT') {
    throw new Error(mintReceipt.result.message);
  }

  expect(mintReceipt.result.state).toEqual('HALT');
  expect(mintReceipt.result.value).toBeUndefined();
  expect(mintReceipt.result.gasConsumed).toMatchSnapshot('mint consumed');
  expect(mintReceipt.events).toHaveLength(1);
  const event = mintReceipt.events[0];
  expect(event.name).toEqual('transfer');
  if (event.name !== 'transfer') {
    throw new Error('For TS');
  }
  expect(event.parameters.from).toBeUndefined();
  expect(event.parameters.to).toEqual(accountID.address);
  expect(event.parameters.amount.toString()).toEqual('100');
};

const verifySmartContracts = async (
  ico: any,
  token: any,
  escrow: any,
  accountID: UserAccountID,
  toAccountID: UserAccountID,
  nowSeconds: number,
  developerClient?: DeveloperClient,
): Promise<void> => {
  const [
    name,
    symbol,
    decimals,
    amountPerNEO,
    icoOwner,
    startTimeSeconds,
    icoDurationSeconds,
    initialTotalSupply,
    [initialRemaining, initialBalance],
  ] = await Promise.all([
    token.name(),
    token.symbol(),
    token.decimals(),
    ico.amountPerNEO(),
    ico.owner(),
    ico.startTimeSeconds(),
    ico.icoDurationSeconds(),
    token.totalSupply(),
    Promise.all([ico.remaining(), token.balanceOf(accountID.address)]),
  ]);
  expect(name).toEqual('One');
  expect(symbol).toEqual('ONE');
  expect(decimals.toString()).toEqual('8');
  expect(amountPerNEO.toString()).toEqual('10');
  expect(icoOwner).toEqual(accountID.address);
  expect(startTimeSeconds.gte(new BigNumber(nowSeconds))).toBeTruthy();
  expect(icoDurationSeconds.toString()).toEqual('157700000');
  expect(initialTotalSupply.toString()).toEqual('0');
  expect(initialRemaining.toString()).toEqual(new BigNumber(10_000_000_000).toString());
  expect(initialBalance.toString()).toEqual('0');

  await mintTokens(ico, accountID, developerClient);

  await verifySmartContractAfterMint(ico, token, escrow, accountID, toAccountID, developerClient);
};

const verifySmartContractTesting = async (codegenPath: string, nowSeconds: number) => {
  // tslint:disable-next-line no-require-imports
  const test = require(nodePath.resolve(codegenPath, 'test'));
  await test.withContracts(async ({ ico, token, escrow, masterAccountID, networkName, client }: any) => {
    await client.providers.memory.keystore.addUserAccount({
      network: networkName,
      privateKey: TO_PRIVATE_KEY,
    });
    await verifySmartContracts(
      ico,
      token,
      escrow,
      masterAccountID,
      { network: networkName, address: privateKeyToAddress(TO_PRIVATE_KEY) },
      nowSeconds,
    );
  });
};

const getGeneratedICOCode = (
  dataRoot: string,
): {
  readonly abi: ABI;
  readonly contract: {
    readonly createSmartContract: (client: Client) => any;
  };
} => {
  // tslint:disable-next-line no-require-imports
  const abi = require(nodePath.resolve(dataRoot, 'ICO', 'abi')).icoABI;
  // tslint:disable-next-line no-require-imports
  const contract = require(nodePath.resolve(dataRoot, 'ICO', 'contract'));
  const createSmartContract = contract.createICOSmartContract;

  return { abi, contract: { createSmartContract } };
};

const getGeneratedTokenCode = (
  dataRoot: string,
): {
  readonly abi: ABI;
  readonly contract: {
    readonly createSmartContract: (client: Client) => any;
  };
} => {
  // tslint:disable-next-line no-require-imports
  const abi = require(nodePath.resolve(dataRoot, 'Token', 'abi')).tokenABI;
  // tslint:disable-next-line no-require-imports
  const contract = require(nodePath.resolve(dataRoot, 'Token', 'contract'));
  const createSmartContract = contract.createTokenSmartContract;

  return { abi, contract: { createSmartContract } };
};

const getGeneratedEscrowCode = (
  dataRoot: string,
): {
  readonly abi: ABI;
  readonly contract: {
    readonly createSmartContract: (client: Client) => any;
  };
} => {
  // tslint:disable-next-line no-require-imports
  const abi = require(nodePath.resolve(dataRoot, 'Escrow', 'abi')).escrowABI;
  // tslint:disable-next-line no-require-imports
  const contract = require(nodePath.resolve(dataRoot, 'Escrow', 'contract'));
  const createSmartContract = contract.createEscrowSmartContract;

  return { abi, contract: { createSmartContract } };
};

const getGeneratedCommonCode = (
  dataRoot: string,
): {
  readonly createClient: () => Client;
  readonly createDeveloperClients: () => { readonly [network: string]: DeveloperClient };
  // tslint:disable-next-line no-require-imports
} => require(nodePath.resolve(dataRoot, 'client'));

const verifySmartContractsManual = async (
  dataRoot: string,
  accountID: UserAccountID,
  toAccountID: UserAccountID,
  nowSeconds: number,
) => {
  const {
    abi: icoABI,
    contract: { createSmartContract: createICOSmartContract },
  } = getGeneratedICOCode(dataRoot);
  expect(icoABI).toBeDefined();
  const {
    abi: tokenABI,
    contract: { createSmartContract: createTokenSmartContract },
  } = getGeneratedTokenCode(dataRoot);
  expect(tokenABI).toBeDefined();
  const {
    abi: escrowABI,
    contract: { createSmartContract: createEscrowSmartContract },
  } = getGeneratedEscrowCode(dataRoot);
  expect(escrowABI).toBeDefined();

  const { createClient, createDeveloperClients } = getGeneratedCommonCode(dataRoot);
  const client = createClient();
  const developerClient = createDeveloperClients().local;

  const ico = createICOSmartContract(client);
  const token = createTokenSmartContract(client);
  const escrow = createEscrowSmartContract(client);

  await client.providers.memory.keystore.addUserAccount({
    network: accountID.network,
    privateKey: TO_PRIVATE_KEY,
  });

  await verifySmartContracts(ico, token, escrow, accountID, toAccountID, nowSeconds, developerClient);
};

const verifySmartContractAfterMint = async (
  ico: any,
  token: any,
  escrow: any,
  accountID: UserAccountID,
  toAccountID: UserAccountID,
  developerClient?: DeveloperClient,
): Promise<void> => {
  const [totalSupply, remaining, balance, toBalance] = await Promise.all([
    token.totalSupply(),
    ico.remaining(),
    token.balanceOf(accountID.address),
    token.balanceOf(toAccountID.address),
  ]);
  expect(totalSupply.toString()).toEqual('100');
  expect(remaining.toString()).toEqual(new BigNumber(9_999_999_900).toString());
  expect(balance.toString()).toEqual('100');
  expect(toBalance.toString()).toEqual('0');

  const result = await token.transfer(accountID.address, toAccountID.address, new BigNumber('25'));
  const [receipt] = await Promise.all([
    result.confirmed({ timeoutMS: 2500 }),
    developerClient === undefined ? Promise.resolve() : developerClient.runConsensusNow(),
  ]);
  if (receipt.result.state === 'FAULT') {
    throw new Error(receipt.result.message);
  }

  const [totalSupplyAfter, balanceAfter, toBalanceAfter] = await Promise.all([
    token.totalSupply(),
    token.balanceOf(accountID.address),
    token.balanceOf(toAccountID.address),
  ]);
  expect(totalSupplyAfter.toString()).toEqual('100');
  expect(balanceAfter.toString()).toEqual('75');
  expect(toBalanceAfter.toString()).toEqual('25');

  const escrowAddress = escrow.definition.networks[accountID.network].address;
  const [escrowTransferReceipt] = await Promise.all([
    token.transfer.confirmed(
      accountID.address,
      escrowAddress,
      new BigNumber('25'),
      ...escrow.forwardApproveReceiveTransferArgs(toAccountID.address),
    ),
    developerClient === undefined ? Promise.resolve() : developerClient.runConsensusNow(),
  ]);
  if (escrowTransferReceipt.result.state === 'FAULT') {
    throw new Error(escrowTransferReceipt.result.message);
  }
  expect(escrowTransferReceipt.events).toHaveLength(2);
  const transferEvent = escrowTransferReceipt.events[1];
  if (transferEvent.name !== 'transfer') {
    throw new Error('Expected transfer event');
  }
  expect(transferEvent.parameters.amount.toString()).toEqual('25');
  const balanceAvailableEvent = escrowTransferReceipt.events[0];
  if (balanceAvailableEvent.name !== 'balanceAvailable') {
    throw new Error('Expected balanceAvailable event');
  }
  expect(balanceAvailableEvent.parameters.amount.toString()).toEqual('25');

  const tokenAddress = token.definition.networks[accountID.network].address;
  const [escrowBalanceAfterEscrow, balanceAfterEscrow, toBalanceAfterEscrow, escrowPairBalance] = await Promise.all([
    token.balanceOf(escrowAddress),
    token.balanceOf(accountID.address),
    token.balanceOf(toAccountID.address),
    escrow.balanceOf(accountID.address, toAccountID.address, tokenAddress),
  ]);
  expect(escrowBalanceAfterEscrow.toString()).toEqual('25');
  expect(balanceAfterEscrow.toString()).toEqual('50');
  expect(toBalanceAfterEscrow.toString()).toEqual('25');
  expect(escrowPairBalance.toString()).toEqual('25');

  const [escrowClaimReceipt] = await Promise.all([
    escrow.claim.confirmed(accountID.address, toAccountID.address, tokenAddress, new BigNumber('10'), {
      from: toAccountID,
    }),
    developerClient === undefined ? Promise.resolve() : developerClient.runConsensusNow(),
  ]);
  if (escrowClaimReceipt.result.state === 'FAULT') {
    throw new Error(escrowClaimReceipt.result.message);
  }
  expect(escrowClaimReceipt.result.value).toEqual(true);
  expect(escrowClaimReceipt.events).toHaveLength(2);

  const [escrowBalanceAfterClaim, balanceAfterClaim, toBalanceAfterClaim, escrowPairBalanceAfterClaim] =
    await Promise.all([
      token.balanceOf(escrowAddress),
      token.balanceOf(accountID.address),
      token.balanceOf(toAccountID.address),
      escrow.balanceOf(accountID.address, toAccountID.address, tokenAddress),
    ]);
  expect(escrowBalanceAfterClaim.toString()).toEqual('15');
  expect(balanceAfterClaim.toString()).toEqual('50');
  expect(toBalanceAfterClaim.toString()).toEqual('35');
  expect(escrowPairBalanceAfterClaim.toString()).toEqual('15');
};

export const buildTests = async (project: string) => {
  const start = Math.round(Date.now() / 1000);
  const exec = one.createExec(project);
  one.addCleanup(async () => {
    await exec('stop network');
  });
  await exec('build');

  const [{ client: outerClient }, config] = await Promise.all([getClients(project), one.getProjectConfig(project)]);
  const contracts = await getContracts(outerClient, constants.LOCAL_NETWORK_NAME);
  verifyICOContract(contracts.find((contract) => contract.manifest.name === 'ICO'));
  verifyTokenContract(contracts.find((contract) => contract.manifest.name === 'Token'));
  verifyEscrowContract(contracts.find((contract) => contract.manifest.name === 'Escrow'));

  await Promise.all([
    verifySmartContractTesting(config.codegen.path, start),
    verifySmartContractsManual(
      config.codegen.path,
      { network: constants.LOCAL_NETWORK_NAME, address: privateKeyToAddress(constants.PRIVATE_NET_PRIVATE_KEY) },
      { network: constants.LOCAL_NETWORK_NAME, address: privateKeyToAddress(TO_PRIVATE_KEY) },
      start,
    ),
  ]);
};
