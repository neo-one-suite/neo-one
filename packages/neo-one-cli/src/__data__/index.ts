export * from './getClients';
export * from './getContracts';
export * from './buildTests';
export * from './generateTests';
