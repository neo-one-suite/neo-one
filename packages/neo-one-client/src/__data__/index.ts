export * from './keys';
export * from './transactions';
