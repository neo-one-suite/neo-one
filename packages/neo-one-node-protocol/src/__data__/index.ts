export { context } from './context';
export { keys } from './keys';
