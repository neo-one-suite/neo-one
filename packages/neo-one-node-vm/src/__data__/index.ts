import * as assets from './assets';
import * as transactions from './transactions';
import * as testUtils from './utils';

export * from './createBlockchain';
export * from './keys';

export { assets, testUtils, transactions };
