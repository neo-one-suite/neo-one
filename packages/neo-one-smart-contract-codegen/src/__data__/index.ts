export * from './nep17';
export * from './testUtils';
export * from './abiFactory';
