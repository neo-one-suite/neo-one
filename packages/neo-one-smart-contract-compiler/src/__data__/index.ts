import * as helpers from './helpers';

export { helpers };
export * from './keys';
