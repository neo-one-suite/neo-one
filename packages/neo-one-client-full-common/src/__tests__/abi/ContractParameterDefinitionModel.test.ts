import { contractParamDefinitionModel } from '../../__data__';

describe('ContractParameterModel - serializeJSON', () => {
  test('Boolean', () => {
    expect(contractParamDefinitionModel.boolean.serializeJSON()).toMatchSnapshot();
  });

  test('ByteArray', () => {
    expect(contractParamDefinitionModel.byteArray.serializeJSON()).toMatchSnapshot();
  });

  test('Hash160', () => {
    expect(contractParamDefinitionModel.hash160.serializeJSON()).toMatchSnapshot();
  });

  test('Hash256', () => {
    expect(contractParamDefinitionModel.hash256.serializeJSON()).toMatchSnapshot();
  });

  test('Array', () => {
    expect(contractParamDefinitionModel.array.serializeJSON()).toMatchSnapshot();
  });

  test('Integer', () => {
    expect(contractParamDefinitionModel.integer.serializeJSON()).toMatchSnapshot();
  });

  test('InteropInterface', () => {
    expect(contractParamDefinitionModel.interopInterface.serializeJSON()).toMatchSnapshot();
  });

  test('Map', () => {
    expect(contractParamDefinitionModel.map.serializeJSON()).toMatchSnapshot();
  });

  test('PublicKey', () => {
    expect(contractParamDefinitionModel.publicKey.serializeJSON()).toMatchSnapshot();
  });

  test('Signature', () => {
    expect(contractParamDefinitionModel.signature.serializeJSON()).toMatchSnapshot();
  });

  test('String', () => {
    expect(contractParamDefinitionModel.string.serializeJSON()).toMatchSnapshot();
  });

  test('Void', () => {
    expect(contractParamDefinitionModel.void.serializeJSON()).toMatchSnapshot();
  });
});
