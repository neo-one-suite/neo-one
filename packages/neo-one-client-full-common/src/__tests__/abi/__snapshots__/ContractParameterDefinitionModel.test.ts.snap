// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractParameterModel - serializeJSON Array 1`] = `
Object {
  "name": "param",
  "type": "Array",
}
`;

exports[`ContractParameterModel - serializeJSON Boolean 1`] = `
Object {
  "name": "param",
  "type": "Boolean",
}
`;

exports[`ContractParameterModel - serializeJSON ByteArray 1`] = `
Object {
  "name": "param",
  "type": "ByteArray",
}
`;

exports[`ContractParameterModel - serializeJSON Hash160 1`] = `
Object {
  "name": "param",
  "type": "Hash160",
}
`;

exports[`ContractParameterModel - serializeJSON Hash256 1`] = `
Object {
  "name": "param",
  "type": "Hash256",
}
`;

exports[`ContractParameterModel - serializeJSON Integer 1`] = `
Object {
  "name": "param",
  "type": "Integer",
}
`;

exports[`ContractParameterModel - serializeJSON InteropInterface 1`] = `
Object {
  "name": "param",
  "type": "InteropInterface",
}
`;

exports[`ContractParameterModel - serializeJSON Map 1`] = `
Object {
  "name": "param",
  "type": "Map",
}
`;

exports[`ContractParameterModel - serializeJSON PublicKey 1`] = `
Object {
  "name": "param",
  "type": "PublicKey",
}
`;

exports[`ContractParameterModel - serializeJSON Signature 1`] = `
Object {
  "name": "param",
  "type": "Signature",
}
`;

exports[`ContractParameterModel - serializeJSON String 1`] = `
Object {
  "name": "param",
  "type": "String",
}
`;

exports[`ContractParameterModel - serializeJSON Void 1`] = `
Object {
  "name": "param",
  "type": "Void",
}
`;
